### ISSUE TITLE
Current blocTest implementation doesn't allow to have a seed calculated in runtime 

### ISSUE THREAD

**Issue by solikhver** [felangel/bloc#2206]
Hi Felix. 
Thank you for your great work.

I am trying to use a complex object seed generated during async setUp.

I propose to replace `State seed` in `runBlocTest` with  `State Function() seed` function. 

Here is example that doesn't work in current version.
```
  group('CounterBloc', () {
    var mySeed = 0;

    setUp(() {
      mySeed = 2;
    });

    blocTest<CounterBloc, int>(
      'emits [3] when CounterEvent.increment is added',
      build: () => CounterBloc(),
      seed: mySeed,
      act: (bloc) async => bloc.add(CounterEvent.increment),
      expect: const <int>[3],
    );
  });
```
Here is my solution: https://github.com/solikhver/bloc/tree/seed_function_in_bloc_test
I can create pull request if my proposition will be approved. 

Best regards.

**Comment by felangel** [comment]
Hi @solikhver 👋 
Thanks for opening an issue!

Yeah I agree we might as well make this change for consistency with the remaining APIs in the null-safety prerelease of bloc_test. I can apply them since it's late in Germany and you're probably asleep 😄 

Thanks so much for bringing this up and I'll try to get it released in the next few hours 👍

**Comment by felangel** [comment]
This is now available as part of [v8.0.0-nullsafety.4](https://pub.dev/packages/bloc_test/versions/8.0.0-nullsafety.4) 🎉
